mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BucketNameGenerator.java

```diff
@@ -22,23 +22,32 @@
 import java.time.Clock;
 import java.time.ZonedDateTime;
 import java.time.temporal.ChronoField;
+import java.util.concurrent.ConcurrentLinkedQueue;
 
 import javax.inject.Inject;
 
 import org.apache.james.blob.api.BucketName;
 
 public class BucketNameGenerator {
     private final Clock clock;
+    private final ConcurrentLinkedQueue<BucketName> bucketNames;
 
     @Inject
     public BucketNameGenerator(Clock clock) {
         this.clock = clock;
+        this.bucketNames = new ConcurrentLinkedQueue<>();
     }
 
     public BucketName currentBucket() {
         ZonedDateTime now = ZonedDateTime.now(clock);
         int month = now.getMonth().get(ChronoField.MONTH_OF_YEAR);
         int year = now.getYear();
-        return BucketName.of(String.format("deleted-messages-%d-%d-01", year, month));
+        BucketName bucketName = BucketName.of(String.format("deleted-messages-%d-%d-01", year, month));
+        bucketNames.add(bucketName);
```

Comment:
what is the plan for the non-temporary solution?

Reply:
We can read the buckets used for the vault using the DeletedMessageMetadataVault I guess ^^